### PR TITLE
feat: add wasm-miniscript module

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Wrap the `rust-miniscript` library in a WebAssembly module.

This is just a rough skeleton making sure the CI pipeline works.

Issue: BTC-1317

### CODE REVIEW

**Review by saravanan7mani — DISMISSED** [review]

**Review by zahin-mohammad — CHANGES_REQUESTED** [review]
Can you please update the relevant readme's to let devs know they need to install rust.

From my understanding looks like we are directly committing the source rust code, and generating the wasm at build time?

File: modules/wasm-miniscript/test/descriptorFixtures.ts [hunk 1]
```diff
@@ -0,0 +1,669 @@
+// Copyright (c) 2023 Jose-Luis Landabaso - https://bitcoinerlab.com
+// Distributed under the MIT software license
+
+//This file is generated automatically. Do not edit it!
```
[the remark this answers is not in the document]
Comment (zahin-mohammad) [reply modules/wasm-miniscript/test/descriptorFixtures.ts:4]: Any notice we should add for this code snippet? Do we need to make this specific module under the MIT license?

[the remark this answers is not in the document]
Comment (OttoAllmendinger) [reply modules/wasm-miniscript/test/descriptorFixtures.ts:4]: good idea

**Comment by OttoAllmendinger** [comment]
moving to separate repo

**Pull request closed without merging**